src/ucp/core/ucp_worker.c

```diff
@@ -496,7 +510,7 @@ ucs_status_t ucp_worker_create(ucp_context_h context, ucs_thread_mode_t thread_m
     }
 
     /* Create the underlying UCT worker */
-    status = uct_worker_create(&worker->async, thread_mode, &worker->uct);
+    status = uct_worker_create(&worker->async, worker_params->thread_mode, &worker->uct);
```

Comment:
you have to check params->field_mask before accessing every field - that was the purpose of adding field_mask. if the bit is not set, need to use a default value.